lib/Frontend/ArgsToFrontendInputsConverter.cpp

```diff
@@ -150,3 +160,11 @@ bool ArgsToFrontendInputsConverter::checkForMissingPrimaryFiles(
   }
   return !primaryFiles.empty();
 }
+
+bool ArgsToFrontendInputsConverter::isSingleThreadedWMO() const {
+  Optional<std::vector<std::string>> userSuppliedNamesOrErr =
+      OutputFilesComputer::getOutputFilenamesFromCommandLineOrFilelist(Args,
```

Comment:
AFAICT, it is called thrice. The information could be precomputed and stored somewhere, at the cost of adding some complexity to pass it around from where stored to where consumed. Since the cost is dominated by everything else, I tend to favor the recomputation alternative. But if you prefer, I'm ok to change it.